src/client/startup_screen.h

```diff
@@ -0,0 +1,115 @@
+/*
+Minetest
+Copyright (C) 2020 Pierre-Yves Rollo <dev@pyrollo.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include <string>
+
+class Clouds;
+class MenuTextureSource;
+namespace irr{namespace scene{class ISceneManager;}}
+
+class StartupScreen;
+extern StartupScreen *g_startup_screen;
```

Comment:
ewwww, why

This should be passed as dependencies into using classes instead

Reply:
This global basically replaced g_menuclouds and g_menucloudsmgr (I guess still a C dev signature).

Five classes uses StartupScreen : `guiEngine`, `Client`, `ClientLauncher`, `Game` and `ModApiMainMenu`, may be to complicated for passing instance.

Some sort of static getStartupScreen() returning a singleton would avoid both global variable and passing instance to each class.